hack/lib/golang.sh

```diff
@@ -416,6 +416,13 @@ kube::golang::set_platform_envs() {
         ;;
     esac
   fi
+
+  # if CC is defined for platform then always enable it
+  ccenv=$(echo "$platform" | awk -F/ '{print "KUBE_" toupper($1) "_" toupper($2) "_CC"}')
+  if [ -n "${!ccenv-}" ]; then 
+    export CGO_ENABLED=1
```

Comment:
orthogonal to this PR but leaving a note somewhere: I still think it's a bit odd that we "globally" toggle CGO_ENABLED=1 instead of per-binary for the binaries where it matters to us. I imagine we have a lot of binaries in the build scripts where the default would be suitable (e.g. linters).